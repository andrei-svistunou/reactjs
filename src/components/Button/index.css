.btn {
  color: #fff;
  padding: 7px 50px;
  border: none;
  border-radius: 2px;
}

.btn-primary {
  background-color: #f55263;
}

.btn--secondary {
  background-color: #444;
}

.btn--small {
  padding: 5px 20px;
  font-size: 12px;
}

.btn-invert {
  background: #fff;
  color: #f55263;
}
