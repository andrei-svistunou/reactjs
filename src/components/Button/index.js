import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Button = props => (
  <input type="button" className={`btn ${props.classes} upper-text`} value={props.value} onClick={props.onClick}/>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  classes: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

export default Button;
