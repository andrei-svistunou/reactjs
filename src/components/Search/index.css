.search__label {
  color: #fff;
  display: block;
  padding: 15px 0;
}

.search__field {
  background-color: #000;
  color: #fff;
  border: none;
  border-bottom: 2px solid #cf626f;
  display: block;
  width: 100%;
  padding: 10px;
  margin-bottom: 15px;
}