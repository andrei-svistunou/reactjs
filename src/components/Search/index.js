import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Button from '../Button';
import Filter from '../Filter';
import searchFilter from '../../redux/actions/searchFilter';
import './index.css';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchFieldValue: '',
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChangeInput = this.onChangeInput.bind(this);
  }

  static propTypes = {
    searchFilter: PropTypes.func.isRequired,
    searchingFilters: PropTypes.array.isRequired,
  }

  onChangeInput(e) {
    const { value } = e.target;
    this.setState({ searchFieldValue: value });
  }

  onSubmit(e) {
    e.preventDefault();
    return this.props.clickHandler({
      searchBy: this.state.searchBy.name,
      search: this.state.searchFieldValue
    });
  }

  render() {
    const { searchFieldValue } = this.state;
    const { searchingFilters, searchFilter } = this.props;

    return (
      <form onSubmit={this.onSubmit}>
        <label className="search__label upper-text" htmlFor={this.props.id}>Find your movie</label>
        <input type="text" className="search__field" id={this.props.id}
          value={ searchFieldValue} placeholder="Enter film or author"
          onChange={this.onChangeInput}/>
        <div className="flex">
          <Filter label="Search by" selectFilter={searchFilter} filters={searchingFilters}/>
          <Button value="Search" classes="btn-primary upper main-search"
            onClick={() => this.props.clickHandler({
              searchBy: searchingFilters.find(filter => filter.active).name,
              search: searchFieldValue
            })}/>
        </div>
      </form>
    );
  }
}

Search.propTypes = {
  id: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired
};

const mapStateToProps = state => state.films;
const mapDispatchToProps = (dispatch) => {
  const actions = bindActionCreators({ searchFilter }, dispatch);
  return { ...actions };
};

export default connect(mapStateToProps, mapDispatchToProps)(Search);
