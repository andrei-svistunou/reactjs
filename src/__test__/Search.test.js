import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configStore from '../redux/configureStore';
import Button from '../components/Button';
import Search from '../components/Search';


const initialState = window.REDUX_INITIAL_STATE || {};
const store = configStore(initialState);

describe('<Search />', () => {
  const Props = {
    id: 'id',
    clickHandler: jest.fn()
  };

  test('should render <Search />', () => {
    const wrapper = mount(<Provider store={store}><Search {...Props}/></Provider>);

    expect(wrapper.contains(Button)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle a child click', () => {
    const wrapper = mount(<Provider store={store}><Search {...Props} /></Provider>);
    wrapper.find(Button).at(2).simulate('click');
    expect(Props.clickHandler).toHaveBeenCalled();
  });
});

