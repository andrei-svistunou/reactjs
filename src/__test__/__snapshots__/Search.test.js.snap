// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render <Search /> 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Search)
    clickHandler={[MockFunction]}
    id="id"
  >
    <Search
      clickHandler={[MockFunction]}
      id="id"
      limit={null}
      moviesList={Array []}
      offset={null}
      orderFilters={
        Array [
          Object {
            "active": false,
            "name": "release_date",
            "sortingFilter": "byDate",
            "value": "release date",
          },
          Object {
            "active": false,
            "name": "rating",
            "sortingFilter": "byVote",
            "value": "rating",
          },
        ]
      }
      searchFilter={[Function]}
      searchingFilters={
        Array [
          Object {
            "active": true,
            "name": "title",
            "value": "title",
          },
          Object {
            "active": false,
            "name": "genres",
            "value": "genre",
          },
        ]
      }
      selectedFilm={null}
      total={null}
    >
      <form
        onSubmit={[Function]}
      >
        <label
          className="search__label upper-text"
          htmlFor="id"
        >
          Find your movie
        </label>
        <input
          className="search__field"
          id="id"
          onChange={[Function]}
          placeholder="Enter film or author"
          type="text"
          value=""
        />
        <div
          className="flex"
        >
          <Connect(Filter)
            filters={
              Array [
                Object {
                  "active": true,
                  "name": "title",
                  "value": "title",
                },
                Object {
                  "active": false,
                  "name": "genres",
                  "value": "genre",
                },
              ]
            }
            label="Search by"
            selectFilter={[Function]}
          >
            <Filter
              dispatch={[Function]}
              filters={
                Array [
                  Object {
                    "active": true,
                    "name": "title",
                    "value": "title",
                  },
                  Object {
                    "active": false,
                    "name": "genres",
                    "value": "genre",
                  },
                ]
              }
              label="Search by"
              limit={null}
              moviesList={Array []}
              offset={null}
              orderFilters={
                Array [
                  Object {
                    "active": false,
                    "name": "release_date",
                    "sortingFilter": "byDate",
                    "value": "release date",
                  },
                  Object {
                    "active": false,
                    "name": "rating",
                    "sortingFilter": "byVote",
                    "value": "rating",
                  },
                ]
              }
              searchingFilters={
                Array [
                  Object {
                    "active": true,
                    "name": "title",
                    "value": "title",
                  },
                  Object {
                    "active": false,
                    "name": "genres",
                    "value": "genre",
                  },
                ]
              }
              selectFilter={[Function]}
              selectedFilm={null}
              total={null}
            >
              <div>
                <span
                  className="filter__label upper-text "
                >
                  Search by
                </span>
                <Button
                  classes="btn--small btn-primary left-space-10"
                  key="title"
                  onClick={[Function]}
                  value="title"
                >
                  <input
                    className="btn btn--small btn-primary left-space-10 upper-text"
                    onClick={[Function]}
                    type="button"
                    value="title"
                  />
                </Button>
                <Button
                  classes="btn--small btn--secondary left-space-10"
                  key="genre"
                  onClick={[Function]}
                  value="genre"
                >
                  <input
                    className="btn btn--small btn--secondary left-space-10 upper-text"
                    onClick={[Function]}
                    type="button"
                    value="genre"
                  />
                </Button>
              </div>
            </Filter>
          </Connect(Filter)>
          <Button
            classes="btn-primary upper main-search"
            onClick={[Function]}
            value="Search"
          >
            <input
              className="btn btn-primary upper main-search upper-text"
              onClick={[Function]}
              type="button"
              value="Search"
            />
          </Button>
        </div>
      </form>
    </Search>
  </Connect(Search)>
</Provider>
`;
